calico_mesos/calico_mesos.py

```diff
@@ -71,6 +72,19 @@ def calico_mesos():
     except KeyError:
         raise IsolatorException(ERROR_MISSING_ARGS)
 
+    # Unfortunately, netmodules passes in labels poorly, so move them
+    # around to match the agreed upon design.
+    if args.has_key('labels'):
+        # copy labels so we're not modifiying the original as we iterate
+        _labels = copy.deepcopy(args['labels'])
+        args['labels'] = {}
+        for label in _labels:
+            # Move the key and value into the labels dictionary
+            key = label['key']
```

Comment:
Is a list of key/values not actually better, since it means you have ordering, and you can have a label specified multiple times...

i.e. is the spec just a bit pants?